src/client/components/editor/task-list.js

```diff
@@ -0,0 +1,164 @@
+const DirtyComponent = require('../dirty-component');
+const h = require('react-hyperscript');
+
+const { makeClassList } = require('../../../util');
+
+const Notification = require('../notification');
+const NotificationList = require('../notification/list');
+const NotificationPanel = require('../notification/panel');
+
+const getIncompleteEntities = doc => doc.entities().filter(ent => !ent.completed());
+
+
+const eleEvts = [ 'rename', 'redescribe' ];
+const entEvts = [ 'modify', 'associated', 'unassociated', 'complete', 'uncomplete'];
+const intnEvts = [ 'retype' ];
+
+let bindEleEvts = (ele, cb) => {
+  eleEvts.forEach(evt => {
+
+    ele.on(evt, cb);
+  });
+
+  if( ele.isInteraction() ){
+    intnEvts.forEach(evt => {
+      ele.on(evt, cb);
+    });
+  } else {
+    entEvts.forEach(evt => {
+      ele.on(evt, cb);
+    });
+  }
+};
+
+let unbindEleEvts = (ele, cb) => {
+  eleEvts.forEach(evt => {
+    ele.removeListener(evt, cb);
+  });
+
+  if( ele.isInteraction() ){
+    intnEvts.forEach(evt => {
+      ele.removeListener(evt, cb);
+    });
+  } else {
+    entEvts.forEach(evt => {
+      ele.removeListener(evt, cb);
+    });
+  }
+};
+
+class TaskList extends DirtyComponent {
+  shouldComponentUpdate() {
+    return true;
+  }
+  componentDidMount(){
+    this.eleEvts = eleEvts;
+    this.entEvts = entEvts;
+    this.intnEvts = intnEvts;
+
+    let update = () => this.dirty();
+    this.update = update;
+
+    this.onAdd = ele => {
+      bindEleEvts(ele, update);
+      this.dirty();
+    };
+
+    this.onRemove = ele => {
+      unbindEleEvts(ele, update);
+      this.dirty();
+    };
+
+    this.props.document.on('add', this.onAdd);
+    this.props.document.on('remove', this.onRemove);
+
+    this.props.document.elements().forEach(ele => bindEleEvts(ele, update));
+  }
+
+  componentWillUnmount(){
+    this.props.document.elements().forEach( ele => unbindEleEvts(ele, this.update));
+
+    this.props.document.removeListener(this.onAdd);
+    this.props.document.removeListener(this.onRemove);
+
+  }
+
+  render(){
+    let doc = this.props.document;
+    let ntfns = doc.entities().filter(ent => !ent.completed()).map(ent => {
+      let ntfn = new Notification({
+        openable: true,
+        dismissable: false,
+        openText: 'Show me',
+        active: true,
+        message: `Provide more information for ${ent.name() === '' ? 'unnamed entity' : ent.name()}.`
```

Comment:
Not sure if this would be confusing to users or not: The displayed name won't match up in all cases.  If you have an entity named 'pcna' and then you quit the wizard, your message will show 'pcna' but the graph will show '?'.

Reply:
Maybe

> Provide more information for pcna (?)